extensions/amp-dailymotion/1.0/component.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '#preact';
+import {forwardRef} from '#preact/compat';
+import {useMemo} from '#preact';
+import {VideoEvents} from '../../../src/video-interface';
+import {dispatchCustomEvent} from '#core/dom';
+import {
+  DailymotionEvents,
+  getDailymotionIframeSrc,
+  makeDailymotionMessage,
+} from '../dailymotion-api';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {parseQueryString} from '#core/types/string/url';
+
+/**
+ * @param {string} method
+ * @return {string}
+ */
+function makeMethodMessage(method) {
+  switch (method) {
+    case 'mute':
+      return makeDailymotionMessage('muted', [true]);
+    case 'unmute':
+      return makeDailymotionMessage('muted', [false]);
+    case 'showControls':
+      return makeDailymotionMessage('controls', [true]);
+    case 'hideControls':
+      return makeDailymotionMessage('controls', [false]);
+    default:
+      return makeDailymotionMessage(method);
+  }
+}
+
+/**
+ * @param {MessageEvent} event
+ */
+function onMessage({currentTarget, data}) {
+  const parsedData = parseQueryString(/** @type {string} */ (data));
+  const event = parsedData?.['event'];
+  if (event === DailymotionEvents.API_READY) {
+    dispatchCustomEvent(currentTarget, 'canplay');
+    return;
+  }
+  if (event === DailymotionEvents.END) {
+    dispatchCustomEvent(currentTarget, [VideoEvents.ENDED, VideoEvents.PAUSE]);
+    return;
+  }
+  if (event === DailymotionEvents.PAUSE) {
+    dispatchCustomEvent(currentTarget, VideoEvents.PAUSE);
+    return;
+  }
+  if (event === DailymotionEvents.PLAY) {
+    dispatchCustomEvent(currentTarget, VideoEvents.PLAY);
```

Comment:
We need `playing` event to trigger the autoplaying UI
```suggestion
    dispatchCustomEvent(currentTarget, VideoEvents.PLAYING);
```

Reply:
It may be helpful to have a `DAILYMOTION_EVENTS` to do this conversion for both versions via the dailymotion-api, similar to `VIMEO_EVENTS` and `BRIGHTCOVE_EVENTS`